docs/modules/ROOT/pages/troubleshooting/index.adoc

```diff
@@ -44,4 +44,110 @@ If you see this error message, it means that the pipeline run has exceeded the d
+details, see the Tekton link:https://tekton.dev/docs/pipelines/auth/[Authentication] docs. In short:
+
+. Tekton uses a ServiceAccount (named `appstudio-pipeline` by default in Konflux) to run your Pipeline.
+. This ServiceAccount has a list of `imagePullSecrets` and `secrets`.
+.. Tekton uses the `imagePullSecrets` when pulling the images that execute the Tasks in your pipeline.
+   These images are typically hardcoded in the Task definition and publicly accessible. As a Konflux
+   user, you're more interested in the `secrets`.
+.. Tekton injects the `secrets` into the execution environment so that the tools executed inside
+   the Tasks (e.g. `buildah`, `skopeo`, `oras`, `cosign`) can authenticate to registries.
+
+Tekton takes all the `dockercfg` and `dockerconfigjson` secrets linked in `secrets` (as well as
+link:https://tekton.dev/docs/pipelines/auth/#configuring-basic-auth-authentication-for-docker[specially annotated]
+`basic-auth` secrets), merges them into a single file and injects the file into the Task Pod at
+`~/.docker/config.json`. The format of the file is described in the
+link:https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md[containers-auth.json manpage].
+
+Note that the merged file can contain multiple credentials for a single registry, distinguished by
+path. For example:
+
+[source,json]
+----
+{
+  "auths": {
+    "quay.io/my-org": {
+      "auth": "..."
+    },
+    "quay.io": {
+      "auth": "..."
+    }
+  }
+}
+----
+
+The most specific path takes precedence. When accessing images in `quay.io/my-org/`, the tool will
+prefer the `quay.io/my-org` credential over the generic `quay.io` credential (assuming the tool
+implements the `containers-auth.json` spec correctly).
+
+More tips and tricks for debugging below.
+
+=== Check if a secret has access to a registry / to a specific image
+
+[source,bash]
+----
+secret_name=secret-for-my-registry
+should_have_access_to=my.registry.io/my-org/my-image
+
+kubectl get secret "$secret_name" -o json |
+    jq '.data[] | @base64d | fromjson | {auths: (.auths // .)}' |
+    tee /tmp/auth.json
+
+# Check if a tool can use the authfile to access a registry / an image. E.g. skopeo:
+skopeo login --authfile /tmp/auth.json "$should_have_access_to"
+----
+
+Note: works for `dockercfg` and `dockerconfigjson` secrets, not `basic-auth`.
+
+=== Check if the secret is linked to the `appstudio-pipeline` service account
+
+[source,bash]
+----
+kubectl get sa appstudio-pipeline -o json
+----
+
+Does your secret appear in the `secrets` section? If not, link it to the service account with:
+
+[source,bash]
+----
+secret_name=secret-for-my-registry
+
+oc secret link appstudio-pipeline "$secret_name"
+----
+
+=== Get the merged registry auth file
+
+This script roughly approximates Tekton's merging of `dockercfg` and `dockerconfigjson` secrets.
+It does not handle `basic-auth` secrets.
```

Comment:
Does this script have any prerequisites? Platform requirements?

Reply:
`jq`, `kubectl` and read access to Secrets (same as the other script)

Reply:
We have requirements specified for procedures in various files. Can we just add a quick call-out? The CLI and Admin access are likely more interesting pre-requisites.

Reply:
Added